src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/ConditionalOrderPronouncedCoverLetterHelper.java

```diff
@@ -140,4 +191,36 @@ private Map<String, Object> templateVarsForOfflineRespondent(final CaseData case
 
         return templateContent;
     }
+
+    private Map<String, Object> templateVarsForJSSolicitor(final CaseData caseData,
+                                                           final Long caseId,
+                                                           final Applicant applicant) {
+
+        final Map<String, Object> templateContent = docmosisCommonContent.getBasicDocmosisTemplateContent(
+            applicant.getLanguagePreference());
+
+        templateContent.put(SOLICITOR_NAME, applicant.getSolicitor().getName());
+        templateContent.put(SOLICITOR_FIRM, applicant.getSolicitor().getFirmName());
+        templateContent.put(SOLICITOR_ADDRESS, applicant.getSolicitor().getAddress());
+        templateContent.put(DATE, LocalDate.now().format(DATE_TIME_FORMATTER));
+        templateContent.put(CASE_REFERENCE, caseId != null ? formatId(caseId) : null);
```

Comment:
Is this check required? I believe caseId will never be null?

Reply:
You're probably correct, but that's how it's done elsewhere (for some reason).

Reply:
Yeah i dont think this is needed. Maybe done elsewhere from older code but case can't exist without one at this point